src/sagas.js

```diff
@@ -424,19 +443,29 @@ function* fetchVmsSessions ({ vms }) {
   })
 }
 
-function* fetchVmCDRom ({ vmId, running }) {
-  const cdrom = yield callExternalAction('getCDRom', Api.getCDRom, { type: 'GET_VM_CDROM', payload: { vmId, running } })
+/*
+ * Fetch a VM's cdrom configuration based on the status of the VM. A running VM's cdrom
+ * info comes from "current=true" while a non-running VM's cdrom info comes from the
+ * next_run/"current=false" API parameter.
+ */
+function* fetchVmCdRom ({ vmId, current }) {
+  const cdrom = yield callExternalAction('getCdRom', Api.getCdRom, getVmCdRom({ vmId, current }))
+
+  let cdromInternal = null
   if (cdrom) {
-    const cdromInternal = Api.CDRomToInternal({ cdrom })
-    return cdromInternal
+    cdromInternal = Api.cdRomToInternal({ cdrom })
   }
-  return null
+  return cdromInternal
 }
 
-function* fetchVmsCDRom ({ vms }) {
+/*
+ * For each given VM, first fetch the appropriate cdrom info based on the status of the VM,
+ * then push the info to VM reducers for the redux store.
+ */
+function* fetchVmsCdRom ({ vms }) {
   yield * foreach(vms, function* (vm) {
-    const cdromInternal = yield fetchVmCDRom({ vmId: vm.id, running: vm.status === 'up' })
-    yield put(setVmCDRom({ vmId: vm.id, cdrom: cdromInternal }))
+    const cdrom = yield fetchVmCdRom({ vmId: vm.id, isVmUp: vm.status === 'up' })
```

Comment:
`fetchVmCdRom` has parameter `current`, but not `isVmUp`

Reply:
good catch, thanks